src/syntax/gats/gat.jl

```diff
@@ -45,6 +53,39 @@ end
 
 allmethods(m::MethodResolver) = pairs(m.bysignature)
 
+function rename(tag::ScopeTag, renames::Dict{Symbol,Symbol}, m::MethodResolver)
+  MethodResolver(rename(tag, renames, m.bysignature))
+end
+
+function rename(tag::ScopeTag, renames::Dict{Symbol,Symbol}, bysignature::Dict{AlgSorts, Ident})
+  if length(bysignature) == 0
+    bysignature
+  else
+    merge(map(collect(bysignature)) do (algsorts, ident)
+      Dict(rename(tag, renames, algsorts) => rename(tag, renames, ident))
+    end...)
+  end
+end
+
+function reident(reps::Dict{Ident, Ident}, key::Ident, m::MethodResolver)
+  MethodResolver(reident(reps, key, m.bysignature))
+end
+
+function reident(reps::Dict{Ident}, key::Ident, bysignature::Dict{AlgSorts, Ident})
```

Comment:
This seems like just a helper function for `reident(_, _, MethodResolver)`, it shouldn't "claim" the `reident` overload for `reident(_, _, Dict{AlgSorts, Ident})` I think...